ansible/roles-infra/infra-osp-resources-destroy/tasks/main.yml

```diff
@@ -18,10 +19,26 @@
     # create instances, ports, etc. that are outside of the heat
     # stack. If you try to tear the heat stack down without deleting
     # those extra ports, for instance, the heat stack teardown will fail.
-    - include_tasks: heat.yml
+    - name: Delete the Heat template
+      include_tasks: heat.yml
 
-    - when: osp_project_create | bool
-      name: Delete the project
+    - name: Loop through and delete user keypairs
+      when: osp_project_create | bool
+      include_tasks: keypairs.yml
+      # apply:
+      #   environment:
+      #     OS_AUTH_URL: "{{ osp_auth_url }}"
+      #     OS_USERNAME: "{{ osp_auth_username }}"
+      #     OS_PASSWORD: "{{ osp_auth_password }}"
+      #     OS_PROJECT_NAME: "{{ osp_project_name }}"
+      #     OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
+      #     OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
+      loop:
+      - "{{ guid }}-user"
+      - sandbox-{{ guid }}-user
+
```

Comment:
When passing a variable to a whole set of tasks with `loop` + `include_tasks`, it's better to use a dedicated variable:

```yaml
loop_control:
  loop_var: _keypair_owner
```